backend/app/resources/update_user_dto.py

```diff
@@ -1,18 +1,16 @@
+from ..graphql.middleware.validate import validate_userinfo
+
+
 class UpdateUserDTO:
-    def __init__(self, **kwargs):
-        self.first_name = kwargs.get("first_name")
-        self.last_name = kwargs.get("last_name")
-        self.email = kwargs.get("email")
-        self.role = kwargs.get("role")
+    def __init__(self, auth_id, info):
+        self.auth_id = auth_id
+        self.info = info
 
     def validate(self):
         error_list = []
-        if type(self.first_name) is not str:
-            error_list.append("The first_name supplied is not a string.")
-        if type(self.last_name) is not str:
-            error_list.append("The last_name supplied is not a string.")
-        if type(self.email) is not str:
-            error_list.append("The email supplied is not a string.")
-        if type(self.role) is not str:
-            error_list.append("The role supplied is not a string.")
+        if type(self.auth_id) is not str:
+            error_list.append("The auth_id supplied is not a string.")
+
+        error_list = validate_userinfo(self.info, error_list)
```

Comment:
This line should be before the check on auth_id, otherwise it overwrites the value that may have been entered in the array.

Reply:
any validate function that takes in an error_list will only ever append to that error_list. if there are no errors in the validate call, it should return the same error_list that was passed in